### PR TITLE
Remove `with` context on os.scandir

### PR DESCRIPTION
**Pull request by sergeyf**
Makes it compatible with py3.5

### CODE REVIEW

File: src/kitools/data_adapters/synapse/synapse_adapter.py [hunk 1]
```diff
@@ -260,12 +260,13 @@ def _get_dirs_and_files(self, local_path):
         dirs = []
         files = []
 
-        with os.scandir(local_path) as iter:
-            for entry in iter:
-                if entry.is_dir(follow_symlinks=False):
-                    dirs.append(entry)
-                else:
-                    files.append(entry)
+        iter = os.scandir(local_path)
```
Comment (pcstout) [inline src/kitools/data_adapters/synapse/synapse_adapter.py]: We should probably wrap this in a `try`/`finally`
```python
entries = os.scandir(local_path)
try:
    for entry in entries:
        if entry.is_dir(follow_symlinks=False):
            dirs.append(entry)
        else:
            files.append(entry)
finally:
    entries.close()
```

**Review by pcstout — CHANGES_REQUESTED** [review]

File: src/kitools/data_adapters/synapse/synapse_adapter.py (hunk 1 above)
Comment (sergeyf) [reply src/kitools/data_adapters/synapse/synapse_adapter.py]: Gotcha.

**Comment by sergeyf** [comment]
@pcstout done!

**Comment by hafen** [comment]
That's great thanks @sergeyf

**Comment by hafen** [comment]
I'll leave it up to @pcstout to merge

**Review by pcstout — APPROVED** [review]

**Pull request merged**